native/cocos/editor-support/spine/Json.cpp

```diff
@@ -41,9 +41,38 @@ THE SOFTWARE.
 #include <spine/Extension.h>
 #include <spine/Json.h>
 #include <spine/SpineString.h>
+#include <spine/MathUtil.h>
 
 #include <assert.h>
 #include <math.h>
+#ifndef __EMSCRIPTEN__
+#include <stdio.h>
+#endif
+
+#ifdef __EMSCRIPTEN__
+static bool parseHex(const char *str, unsigned int *result_) {
+    auto &result = *result_;
+    result = 0;
+    for (int i = 0; i < 4; ++i) {
+        char c = str[i];
+        if ((c < '0' || c > '9') && (c < 'A' || c > 'F') && (c < 'a' || c > 'f')) {
+            result = 0;
+            return false;
+        }
+
+        result <<= 4;
+        if (c >= '0' && c <= '9') {
+            result |= (c - '0');
+        } else if (c >= 'A' && c <= 'F') {
+            result |= (c - 'A' + 10);
+        } else if (c >= 'a' && c <= 'f') {
+            result |= (c - 'a' + 10);
+        }
+    }
+    return true;
```

Comment:
A simpler version of `sscanf(ptr + 1, "%4x", &uc);` which could make the package size smaller.